src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -47,35 +64,58 @@ Status ledgerFromRequest (
 
     if (!hashValue.empty())
     {
+        if (! hashValue.isString ())
+            return {rpcINVALID_PARAMS, "ledgerHashNotString"};
+
         uint256 ledgerHash;
-        if (hashValue.isString() && ledgerHash.SetHex (hashValue.asString ()))
-            ledger = netOps.getLedgerByHash (ledgerHash);
-        else
+        if (! ledgerHash.SetHex (hashValue.asString ()))
             return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
+           
+        if ((ledger = netOps.getLedgerByHash (ledgerHash)) == nullptr)
+        {
+            auto cl = netOps.getCurrentLedger ();
+            if (cl == nullptr || ! synced (netOps, cl))
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
+        }
     }
     else if (indexValue.isNumeric())
     {
-        ledger = netOps.getLedgerBySeq (indexValue.asInt());
+        if ((ledger = netOps.getLedgerBySeq (indexValue.asInt ())) == nullptr)
+        {
+            auto cl = netOps.getCurrentLedger ();
+            if (cl == nullptr || ! synced (netOps, cl))
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
+        }
     }
     else
     {
-        auto index = indexValue.asString();
-        auto isCurrent = index.empty() || index == "current";
+        auto const index = indexValue.asString ();
+        auto const isCurrent = index.empty() || index == "current";
         if (isCurrent)
             ledger = netOps.getCurrentLedger ();
         else if (index == "closed")
-            ledger = getApp().getLedgerMaster ().getClosedLedger ();
+            ledger = netOps.getClosedLedger ();
         else if (index == "validated")
             ledger = netOps.getValidatedLedger ();
         else
             return {rpcINVALID_PARAMS, "ledgerIndexMalformed"};
 
+        if (ledger == nullptr)
```

Comment:
I haven't examined nor tested that case enough to determine if it will never happen. It might be possible for this code to be called before we create those ledger objects. Whomever originally wrote it thought it could happen. Probably even more so now that it can get called without a network connection, as in when starting up for instance.